spec/cli_helper.rb

```diff
@@ -14,3 +14,9 @@ def capture_stdio(input = nil, &block)
   alias capture_stdout capture_stdio
 end
 
+RSpec.configure do |config|
+  config.after(:each) do
+    # To imitate the initial class load before calling `VimGolf::CLI.start`
+    VimGolf.ui = VimGolf::UI.new
```

Comment:
`ui` is a class attribute and once the class is loaded in the spec helper the class attribute state persists between individual spec runs. so if [this](https://github.com/igrigorik/vimgolf/blob/master/spec/lib/cli_spec.rb#L18) spec runs before [this](https://github.com/igrigorik/vimgolf/blob/master/spec/lib/cli_spec.rb#L8) one `VimGolf.ui` will still be an instance of `VimGolf::UI` from the first spec calling `start`. we have to artificially reset the state after each run